batch/batch/driver/job.py

```diff
@@ -258,7 +258,10 @@ async def make_request():
             await instance.incr_failed_request_count()
             raise
 
-    await retry_transient_errors(make_request)
+    def excl(e: Exception):
+        return isinstance(e, asyncio.TimeoutError)
+
+    await retry_transient_errors_with_exclusions(excl, make_request)
```

Comment:
I could be convinced that this is useful in general, but in this case it seems more straightforward to just do the following. What do you think?
```python3
    async def make_request():
         if instance.state in ('inactive', 'deleted'):
             return
         try:
             async with aiohttp.ClientSession(raise_for_status=True, timeout=aiohttp.ClientTimeout(total=5)) as session:
                 await session.delete(url)
                 await instance.mark_healthy()
         except asyncio.TimeoutError as err:
             return
         except aiohttp.ClientResponseError as err:
            if err.status == 404:
                await instance.mark_healthy()
                return
             await instance.incr_failed_request_count()
             raise

     await retry_transient_errors(make_request)
```

Reply:
It keeps the timeout error syntactically closer to the code that times out which seems good.